.github/workflows/ci.yaml

```
@@ -17,57 +17,62 @@ on:
 jobs:
 
   pre-commit-checks:
-    name: "pre-commit hooks - Python 3.8"
+    name: Pre-commit Checks
     runs-on: ubuntu-latest
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
-      - name: Install Python 3.8
-        uses: actions/setup-python@v5
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
         with:
-          python-version: 3.8
-      - name: Run pre-commit checks
-        uses: pre-commit/action@v3.0.1
+          environments: default lint
+      - name: pre-commit
+        run: pixi run pre-commit-run --color=always --show-diff-on-failure
 
   unit-tests:
-    name: "unit tests - Python ${{ matrix.PYTHON_VERSION }} - SA ${{ matrix.SA_VERSION }}"
+    name: "unit tests"
     runs-on: ubuntu-latest
-    env:
-      CI: True
     strategy:
       fail-fast: true
       matrix:
-        PYTHON_VERSION: ['3.8', '3.9', '3.10']
-        SA_VERSION: ['<2.0', '>=2.0']
+        env:
+          - py38
+          - py39
+          - py310
+          - py311
+          - py312
+          - sa1
+          - sa2
     steps:
       - name: Checkout branch
         uses: actions/checkout@v4
-      - name: Set up micromamba
-        uses: mamba-org/setup-micromamba@f8b8a1e23a26f60a44c853292711bacfd3eac822
+      - name: Set up pixi
+        uses: prefix-dev/setup-pixi@v0.8.1
         with:
-          environment-file: environment.yml
-          cache-downloads: true
-          cache-environment: true
-          cache-environment-key: ${{ runner.os }}-${{ hashFiles('environment.yml') }}-${{ matrix.PYTHON_VERSION }}-${{ matrix.SA_VERSION }}
-          create-args: >-
-            python=${{ matrix.PYTHON_VERSION }}
-            sqlalchemy${{ matrix.SA_VERSION }}
-      - name: Run tests
-        uses: ./.github/actions/pytest
+          environments: ${{ matrix.env }}
+      - run: |
+          pixi run -e ${{ matrix.env }} postinstall
+          pixi run -e ${{ matrix.env }} coverage
```

```suggestion
          pixi run -e ${{ matrix.env }} coverage --color=yes
```

how about adding this to all tests? This will make the ci logs a bit prettier (but also the raw logs harder to read; i personally never use them, though)

---

Sgtm!

---

[b182853](https://github.com/Quantco/datajudge/pull/230/commits/b18285342b66c4dd5c806e4bf2ba952472f95b67)